shared/src/main/java/io/pravega/shared/segment/StreamSegmentNameUtils.java

```diff
@@ -150,4 +193,114 @@ public static String getSegmentChunkName(String segmentName, long offset) {
         Preconditions.checkArgument(!segmentName.contains(OFFSET_SUFFIX), "segmentName is already a SegmentChunk name");
         return segmentName + OFFSET_SUFFIX + Long.toString(offset);
     }
+
+    /**
+     * Method to compute 64 bit segment id which takes primary id and secondary id and composes it as
+     * `msb = secondary` `lsb = primary`.
+     * Primary id identifies the segment container mappeing and primary + secondary uniquely identifies a segment
+     * within a stream.
+     *
+     * @param primaryId primary part of id.
+     * @param secondaryId secondary part of id.
+     * @return segment id which is composed using primary and secondary ids.
+     */
+    public static long computeSegmentId(int primaryId, int secondaryId) {
+        Preconditions.checkArgument(primaryId >= 0);
+        Preconditions.checkArgument(secondaryId >= 0);
+        return (long) secondaryId << 32 | (primaryId & 0xFFFFFFFFL);
+    }
+
+    /**
+     * Method to extract primary id from given segment id.
+     *
+     * @param segmentId segment id.
+     * @return primary part of segment id.
+     */
+    public static int getPrimaryId(long segmentId) {
```

Comment:
I'm not sure this needs to be public.

Reply:
We extract primary id in various places -
1. to determine the mapping
2. to determine duplicates
3. to determine mergable segments